### DISCUSSION TITLE
Running in AMD CPU gives core dumped

### DISCUSSION THREAD

**Discussion by IvanPombo** [InteractiveComputerGraphics/SPlisHSPlasH#302] [Q&A]
Dear community,

Currently, I am having a few problems running the simulator in a machine with AMD CPU and no (GPU). 
We are launching a docker container with SPlisHSPlasH installed, based on the `nvidia/cuda:12.3.1-devel-ubuntu20.04`, and building support both for CPU and GPU simultaneously. The reason for this, is to swiftly change machines with the same dockerfile.

Then, we are launching a simple simulation within the docker container in a CPU only machine. 

When running this workflow in an Intel CPU machine, everything works wonderfully. However, when running the same workflow and simulation we are receiving the following error in the `stderr.txt`:

```
# COMMAND: /SPlisHSPlasH_CPU/bin/SPHSimulator sim_dir/config.json --output-dir output --no-gui

Illegal instruction (core dumped)
```

When running locally on an AMD machine with GPU. Everything also works wonderfully!
Are you guys aware of any bug when running in AMD CPU machines?

**Comment by janbender** [comment]
In fact we also have some AMD CPU machines which we use to perform simulations, so far without any problem. However, we don't have tested this in a docker container.

**Reply by IvanPombo** [reply]
Thanks for the answer @janbender! I think this might be related with the way we are building the image. What is interesting is that in Intel CPU machines works. 

Is there are any ideas for the message `Illegal instruction (core dumped)` we are receiving? Is this error fully related with the machine?

**Reply by janbender** [reply]
Not sure what this message means. Did you try to run the simulator without AVX support?

**Reply by IvanPombo** [reply]
Currently, we have tested in two Intel machines:

Following up with a few more details:

This is how we are building SPlisHSPlasH in our Docker image:
```
RUN cmake \
    -S /SPlisHSPlasH -B /SPlisHSPlasH_CPU/build \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE=/SPlisHSPlasH_CPU/bin \
    -DCMAKE_RUNTIME_OUTPUT_DIRECTORY=/SPlisHSPlasH_CPU/bin \
    -DUSE_PYTHON_BINDINGS=OFF \
    -DBuild_PartioViewer=OFF \
    -DBuild_FoamGenerator=OFF \
    -DBuild_VolumeSampling=OFF \
    -DBuild_MeshSkinning=OFF
```

While the software works in the machine with the following CPU flags (got them from running lscpu):

```
Flags:                          fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse3
                                    6 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constan
                                    t_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pcl
                                    mulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave
                                     avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single s
                                    sbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 sm
                                    ep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflusho
                                    pt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat
                                     avx512_vnni md_clear arch_capabilities
```
It doesn’t work on the machine with the following CPU flags:
```
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse
                                    36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm const
                                    ant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni
                                    pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes x
                                    save avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_sin
                                    gle pti ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 hle avx2 smep b
                                    mi2 erms invpcid rtm rdseed adx smap xsaveopt arat md_clear arch_capa
                                    bilities
```
In the last one, we get a “Illegal instruction” message.


So we are installing in the same way, but one has avx512 support and the other doesn't. Should we configure the simulator without the avx512 in the latter case?
